components/ui/data-table/use-data-table-styles.tsx

```diff
@@ -0,0 +1,56 @@
+import { DataTableColumnDef } from "./data-table";
+
+interface useDataStylesInput<TData, TValue> {
+  columns: DataTableColumnDef<TData, TValue>[];
+}
+
+interface GenerateClassnameForHeaderInput {
+  columnId: string;
+}
+
+interface GenerateClassnameForCellInput {
+  columnId: string;
+}
+
+export const useDataTableStyles = <TData, TValue>(input: useDataStylesInput<TData, TValue>) => {
+  const { columns } = input;
+
+  const generateClassnameForHeader = (input: GenerateClassnameForHeaderInput) => {
+    const { columnId } = input;
+
+    const baseClassnameForHeader = "border-b bg-white";
+    const classnameForLeftColumnFixation = "sticky left-0 border-r";
```

Comment:
Q: Hey Sean, this doesn't really seem like it would support more than one fixed column.

A: Right you are! So there is a way to do multiple columns frozen **I think** but it's a wee bit more complicated than this. I'm looking into it but let's not block on it, I'd rather just ship this and I'll open up a "fun feature we might not use but w/e" PR once I've ironed out how to do that just in case it sends me to CSS hell lol